tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_3.py

```diff
@@ -0,0 +1,179 @@
+        """Test the _structural_class_delta_quality_control method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            50,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            False,
+            quality_control,
+        )
+
+        calculator._structural_class_delta_quality_control()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
+
+        assert calculator.structural_class == expected_structural_class
+        assert calculator.explanation.endswith(expected_explanation)
+
+    @pytest.mark.parametrize(
+        ("concrete_grade", "exposure_classes", "expected_structural_class", "expected_explanation"),
+        [
+            (
+                ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+                ExposureClasses(Carbonation.NA, Chloride.NA, ChlorideSeawater.NA, FreezeThaw.NA, Chemical.NA),
+                4,
+                "(no reduction with respect to concrete grade)",
+            ),
+            (
+                ConcreteMaterial(ConcreteStrengthClass("C30/37")),
+                ExposureClasses(Carbonation.XC1, Chloride.NA, ChlorideSeawater.NA, FreezeThaw.NA, Chemical.NA),
+                3,
+                "(concrete grade >= C30/37)",
+            ),
+            (
+                ConcreteMaterial(ConcreteStrengthClass("C35/45")),
+                ExposureClasses(Carbonation.XC2, Chloride.NA, ChlorideSeawater.NA, FreezeThaw.NA, Chemical.NA),
+                3,
+                "(concrete grade >= C35/45)",
+            ),
+            (
+                ConcreteMaterial(ConcreteStrengthClass("C40/50")),
+                ExposureClasses(Carbonation.NA, Chloride.NA, ChlorideSeawater.XS1, FreezeThaw.NA, Chemical.NA),
+                3,
+                "(concrete grade >= C40/50)",
+            ),
+            (
+                ConcreteMaterial(ConcreteStrengthClass("C45/55")),
+                ExposureClasses(Carbonation.NA, Chloride.XD3, ChlorideSeawater.NA, FreezeThaw.NA, Chemical.NA),
+                3,
+                "(concrete grade >= C45/55)",
+            ),
+        ],
+    )
+    def test_structural_class_delta_concrete_grade(
+        self, concrete_grade: ConcreteMaterial, exposure_classes: ExposureClasses, expected_structural_class: int, expected_explanation: str
+    ) -> None:
+        """Test the _structural_class_delta_concrete_grade method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            exposure_classes=exposure_classes,
+            concrete_material=concrete_grade,
+            design_working_life=50,
+            plate_geometry=False,
+            quality_control=False,
+        )
+
+        calculator._structural_class_delta_concrete_grade()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
```

Comment:
This feels like a workaround. The calculator shouldn't be dependent of a private interface to get the final calculations.